### ISSUE TITLE
Add alternative long-read (nanopore) preprocessing tools

### ISSUE THREAD

**Issue by sofstam** [nf-core/taxprofiler#145]
### Description of feature

Since Porechop is no longer supported, it is maybe useful to investigate [Porechop_ABI](https://github.com/bonsai-team/Porechop_ABI) as alternative.

Or https://github.com/epi2me-labs/pychopper

**Comment by jfy133** [comment]
Would need to investigate putting it on conda...

**Comment by sofstam** [comment]
`https://anaconda.org/bioconda/porechop_abi`

There is a conda environment, I think that this can be added in the next release or what do you think?

**Comment by jfy133** [comment]
Oh my bad, I misunderstood their installation instructions 🤦‍♀️

This is fine for inclusion for first release while we wait for final Bracken/KrakenUniq!

**Comment by sofstam** [comment]
I was thinking if we should only support `porechop_abi` and drop `porechop` at all.

**Comment by sofstam** [comment]
According to their github page: `Note that Porechop_ABI is not designed to handle barcoded sequences adapters. Demultiplexing should be done using standard Porechop commands or other appropriate tools.` It is not equivalent as it does not perform demultiplexing. However, demultiplexing is supported by Guppy and I assume it is currenly preferred. @Midnighter have you worked with Nanopore data?

**Comment by qbonenfant** [comment]
Hi, i'm the main developper of ` porechop_abi`.
If you have any question on the project, feel free to ask me directly, i will be glad to answer.

**Regarding the "equivalent or not" part:**

We "only" added a step between the adapter database read and adapter ressearch in the reads.
![porechop_abi](https://user-images.githubusercontent.com/19302812/199268734-22d514bc-09b1-4ffc-9466-f24171c05bb2.png)

The code base of porechop was left as original as possible, and all commands that used to run on `porechop` are unchanged. The behaviors of the two softwares are identical, as long as you don't use `-abi` or `-go` options.
What `porechop `can do, `porechop_abi `can too.
The part we can't handle (at least for now) is inferring barcoded adapters sequences from the reads only.

What could make a difference is the name of the executable.
We had to change it to avoid installation conflicts and it may need to be changed in pipelines if you want to use our version.

TL;DR: It can work the same, but the name is different.


**On the Demultiplexing part:**
Using `porechop `is prety much obsolete for demultiplexing.
Even the dedicated tool ([Deepbinner](https://github.com/rrwick/Deepbinner) )developped by Ryan Wick (original author of `porechop`) is now deemed too old.
Guppy (Nanopore basecaller) seems to be the "current standard" for demultiplexing.

**Comment by Midnighter** [comment]
> However, demultiplexing is supported by Guppy and it is currenly preferred. @Midnighter have you worked with Nanopore data?

I've only worked with FASTQ files that are the result of running guppy so far. I agree with @jfy133 that we should not include demultiplexing in this pipeline. We don't do it for short reads either.

**Comment by sofstam** [comment]
> I've only worked with FASTQ files that are the result of running guppy so far. I agree with @jfy133 that we should not include demultiplexing in this pipeline. We don't do it for short reads either.

I agree, I was just trying to list the differences between `porechop` and `porechop_abi` , not adding demultiplexing steps.

**Comment by sofstam** [comment]
@qbonenfant Thank you for the detailed description 👍

**Comment by sofstam** [comment]
https://github.com/epi2me-labs/pychopper I add this as comment here.

**Comment by jfy133** [comment]
Or https://github.com/epi2me-labs/pychopper

**Comment by jfy133** [comment]
Or https://github.com/wdecoster/chopper

**Comment by LilyAnderssonLee** [comment]
It seems that `Pychopper` supports ONT long-read sequencing, whereas `chopper` is compatible with both PacBio and ONT sequences. Therefore, it might be more advantageous to utilize `chopper` in this context.

**Comment by sofstam** [comment]
`Pychopper` is for cDNA reads so not useful for us. 

I will have to test first but to my understanding, we can use `porechop_abi `as optional step for adapter trimming and drop `porechop`. 
Regarding `chopper`, it can be added as an alternative to `filtlong`. 

What do you think?
